### PR TITLE
Add 'aws_cloudtrail_service_account' data source

### PR DESCRIPTION
**Pull request by ewbankkit**
New data source `aws_cloudtrail_service_account` to return the account ID and account ARN of the AWS CloudTrail service in the specified region.
See:
* http://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-supported-regions.html

Acceptance tests:
```
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudTrailServiceAccount_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSCloudTrailServiceAccount_ -timeout 120m
=== RUN   TestAccAWSCloudTrailServiceAccount_basic
--- PASS: TestAccAWSCloudTrailServiceAccount_basic (29.16s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	30.209s
```

### CODE REVIEW

File: aws/data_source_aws_cloudtrail_service_account.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+func dataSourceAwsCloudTrailServiceAccountRead(d *schema.ResourceData, meta interface{}) error {
+	region := meta.(*AWSClient).region
+	if v, ok := d.GetOk("region"); ok {
+		region = v.(string)
+	}
+
+	if accid, ok := cloudTrailServiceAccountPerRegionMap[region]; ok {
+		d.SetId(accid)
+		d.Set("arn", fmt.Sprintf("arn:%s:iam::%s:root", meta.(*AWSClient).partition, accid))
```
Comment (Ninir) [inline aws/data_source_aws_cloudtrail_service_account.go]: Would probably be better to use https://github.com/terraform-providers/terraform-provider-aws/blob/master/vendor/github.com/aws/aws-sdk-go/aws/arn/arn.go#L28, thoughts?

File: website/docs/d/cloudtrail_service_account.html.markdown [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+layout: "aws"
+page_title: "AWS: aws_cloudtrail_service_account"
+sidebar_current: "docs-aws-datasource-cloudtrail-service-account"
+description: |-
+  Get AWS CloudTrail Service Account ID for storing trail data in S3.
+---
+
+# aws\_cloudtrail\_service\_account
```
Comment (Ninir) [inline website/docs/d/cloudtrail_service_account.html.markdown]: These antislashes are not needed anymore when writing markdown

File: website/docs/d/cloudtrail_service_account.html.markdown [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+resource "aws_s3_bucket" "bucket" {
+  bucket        = "tf-cloudtrail-logging-test-bucket"
+  force_destroy = true
+
+  policy = <<EOF
+{
+	"Version": "2008-10-17",
+	"Statement": [
+		{
+        			"Sid": "Put bucket policy needed for trails",
```
Comment (Ninir) [inline website/docs/d/cloudtrail_service_account.html.markdown]: Can you fix indentations there? 2 spaces would be really nice :)

Comment (ewbankkit) [reply website/docs/d/cloudtrail_service_account.html.markdown]: Will do.

File: website/docs/d/cloudtrail_service_account.html.markdown (hunk 2 above)
Comment (ewbankkit) [reply website/docs/d/cloudtrail_service_account.html.markdown]: OK, I'll correct.

File: aws/data_source_aws_cloudtrail_service_account.go (hunk 1 above)
Comment (ewbankkit) [reply aws/data_source_aws_cloudtrail_service_account.go]: Agreed.
I added a new `arn.go` file and associated functions in https://github.com/terraform-providers/terraform-provider-aws/pull/1775 which I could use here if that's merged and I rebase.
Also mentioned in https://github.com/terraform-providers/terraform-provider-aws/issues/1783.

Comment (Ninir) [reply aws/data_source_aws_cloudtrail_service_account.go]: Agreed, as it has been merged, should be all ok!

**Comment by ewbankkit** [comment]
Rebased from `master` and squashed commits.
Re-ran acceptance tests:
```
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudTrailServiceAccount_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSCloudTrailServiceAccount_ -timeout 120m
=== RUN   TestAccAWSCloudTrailServiceAccount_basic
--- PASS: TestAccAWSCloudTrailServiceAccount_basic (19.89s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	20.919s
```

**Pull request merged**